1주차/131128_토/131128_고혁진.java

```diff
@@ -0,0 +1,25 @@
+class Solution {
+    public String solution(String X, String Y) {
+        int numx[]={0,0,0,0,0,0,0,0,0,0};
+        int numy[]={0,0,0,0,0,0,0,0,0,0};
+        String answer = "";
+        StringBuilder sb = new StringBuilder();
+        for(int i=0;i<X.length();i++){
+            numx[(int) X.charAt(i) - '0']+=1;
+        }
+        for(int i=0;i<Y.length();i++){
+            numy[(int) Y.charAt(i) - '0']+=1;
+        }
+        for(int i=9;i>=0;i--){
+            int num=0;
+            if(numx[i]!=0&&numy[i]!=0)
+                num=numx[i]>numy[i]?numy[i]:numx[i];
+            for(int k=0;k<num;k++) {
+                sb.append(i);
+            }
+        }
```

Comment:
```suggestion
        for(int i=9;i>=0;i--){
            int num = Math.min(numx[i], numy[i]);
            for(int k=0;k<num;k++) {
                sb.append(i);
            }
        }
```

k<num으로 범위제한이 있어서 num이 0이면 for문을 돌지 않아서 if문으로 걸러주지 않아도 될 것 같아요.
Math.min을 쓰냐 삼항연산자를 쓰냐는 그냥 취향이지만 조금 더 가독성이 좋다고 생각하기 때문에 남깁니다.
저는 이 문제를 map을 이용해서 풀었는데 혁진님 풀이가 더 마음에 드네요.

Reply:
와 대박! 그거까지는 생각하지 못했네요 감사합니다!